projects/gateway2/wellknown/delegation.go

```diff
@@ -5,7 +5,8 @@ const (
 	RouteDelegationLabelSelector = "delegation.gateway.solo.io/label"
 
 	// RouteDelegationLabelSelectorWildcard wildcards the namespace to select delegatee routes by label
-	RouteDelegationLabelSelectorWildcardNamespace = "*"
+	// Note: this must be a valid RFC 1123 DNS label
+	RouteDelegationLabelSelectorWildcardNamespace = "all"
```

Comment:
since this makes it not possible to use the delegation label with an actual namespace called `all`, what do you think of instead creating a different label to use for all namespaces, e.g. `delegation.gateway.solo.io/all-namespaces-label` and that would imply we search for the label in all namespaces?
```
   backendRefs:
   - group: delegation.gateway.solo.io
     kind: all-namespaces-label
     name: foobar
     namespace: # this should not be provided; if provided, we ignore (or error)
```
would mean: delegate to routes with label `delegation.gateway.solo.io/all-namespaces-label:foobar` in all namespaces

although i'm not sure if this goes against the [spec](https://gateway-api.sigs.k8s.io/reference/spec/#httpbackendref) which says that empty namespace == local namespace

Reply:
This is just confusing. Let's stick to this and provide a label-override (via Env) when needed. I want see how many customers actually have a namespace called `all` and if they have a problem renaming it.

Reply:
I enabled an Env override.